src/row-data-utils.ts

```diff
@@ -161,37 +163,12 @@ class RowDataUtils {
     const callback =
       typeof optionsOrCallback === 'function' ? optionsOrCallback : cb!;
 
-    if (!rules || (rules as Rule[]).length === 0) {
-      throw new Error('At least one rule must be provided.');
-    }
-
-    rules = arrify(rules).map(rule => {
-      const column = Mutation.parseColumnName(rule.column);
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      const ruleData: any = {
-        familyName: column.family,
-        columnQualifier: Mutation.convertToBytes(column.qualifier!),
-      };
-
-      if (rule.append) {
-        ruleData.appendValue = Mutation.convertToBytes(rule.append);
-      }
-
-      if (rule.increment) {
-        ruleData.incrementAmount = rule.increment;
-      }
-
-      return ruleData;
+    const reqOpts = getRMWRRequest({
+      reqOpts: properties.reqOpts,
+      id: properties.requestData.id,
+      rules,
+      appProfileId: properties.requestData.bigtable.appProfileId,
     });
```

Comment:
All the code removed in this file now lives in getRMWRRequest and one call to getRMWRRequest will be made to build the request.